### PR TITLE
feat(framework) Add alpine base image

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

- Adds Alpine Docker base image.

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Update the changelog entry below
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/docker/base/ubuntru/Dockerfile
Comment (chongshenng) [inline src/docker/base/ubuntru/Dockerfile:1]: I did a quick scan and saw that you had `ubuntru` as the directory. Should that be `ubuntu` instead?

File: src/docker/base/alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# https://pythonspeed.com/articles/multi-stage-docker-python/
+
+ARG PYTHON_VERSION=3.11
+ARG ALPINE_VERSION=alpine3.19
+FROM python:${PYTHON_VERSION}-${ALPINE_VERSION} as compile
+
+# Install system dependencies
+RUN apk add --no-cache \
+    # require for compiling grpcio on ARM64
+    g++=13.2.1_git20231014-r0 \
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply src/docker/base/alpine/Dockerfile:19]: Afaik Dependabot doesn't check these kind of packages. Renovatebot also doesn't have native support, but there are some options: https://github.com/renovatebot/renovate/issues/3717#issuecomment-792652444

File: src/docker/base/alpine/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# Building the base image for ARM64 requires installing some additional system dependencies to
+# compile the grpcio package, as they do not provide a pre-built package. However, we don't want
+# the dependencies in the final base image as they are only needed to compile the package.
+# That's why we're creating a multi-stage build. When installing the flwr, we are using a
+# virtual environment to keep all files in a single isolated directory as described here:
+# https://pythonspeed.com/articles/multi-stage-docker-python/
+
+ARG PYTHON_VERSION=3.11
+ARG ALPINE_VERSION=alpine3.19
+FROM python:${PYTHON_VERSION}-${ALPINE_VERSION} as compile
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply src/docker/base/alpine/Dockerfile]: The slim version is only available for debian.

File: doc/source/contributor-how-to-build-docker-images.rst [hunk 3]
```diff
@@ -66,7 +66,7 @@ The following example creates a base image with Python 3.11.0, pip 23.0.1 and se
 
 .. code-block:: bash
 
-  $ cd src/docker/base/
+  $ cd src/docker/base/ubuntu
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply doc/source/contributor-how-to-build-docker-images.rst:69]: That make sense! I will update the in the followup PR when I'm changing the base Ubuntu base image.

File: src/docker/base/alpine/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Moep90) [reply src/docker/base/alpine/Dockerfile:32]: Ok, if they don't have it, I suggest removing explicit package versions.

File: .github/workflows/docker-base.yml [hunk 4]
```diff
@@ -46,15 +41,19 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11"]
+        os:
+          - tag: ubuntu22.04
```
Comment (Moep90) [inline .github/workflows/docker-base.yml:45]: The tag could be built by os and os_version, similar to the file-dir if its technically possible by github
```
os: ubuntu|alpine
os_version: 22.04|3.19
tag: ${{ matrix.os }}${{ matrix.os_version }}
file-dir: src/docker/base/${{ matrix.os }}
```

Comment (Robert-Steiner) [reply .github/workflows/docker-base.yml:45]: That is unfortunately not possible in github workflows.

File: src/docker/base/alpine/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply src/docker/base/alpine/Dockerfile]: I removed them

**Review by Moep90 — APPROVED** [review]

**Review by tanertopal — APPROVED** [review]
lgtm

**Pull request merged**